components/agentSearchForm/index.jsx

```diff
@@ -22,34 +22,38 @@
 import React, { useState } from "react";
 import { useId } from "react-id-generator";
 import T from "prop-types";
-import { Button, Input } from "@inrupt/prism-react-components";
+import { Form, Button, Input } from "@inrupt/prism-react-components";
 
-export function handleClick({ setAgentId, onSubmit }) {
-  return (agentId) => {
-    setAgentId("");
+function AgentSearchForm({ children, onSubmit, buttonText }) {
+  const [agentId, setAgentId] = useState("");
+  const inputId = useId();
+
+  const handleSubmit = () => {
     onSubmit(agentId);
+    setAgentId("");
   };
-}
 
-export function handleChange(setAgentId) {
-  return ({ target: { value } }) => {
-    setAgentId(value);
+  const handleChange = (event) => {
+    setAgentId(event.target.value);
   };
-}
 
-function AgentSearchForm({ children, onSubmit, buttonText }) {
-  const [agentId, setAgentId] = useState("");
-  const onClick = handleClick({ setAgentId, onSubmit });
-  const onChange = handleChange(setAgentId);
-  const inputId = useId();
   return (
-    <>
-      <Input id={inputId} label="WebID" onChange={onChange} value={agentId} />
+    <Form onSubmit={handleSubmit}>
+      <Input
+        id={inputId}
+        label="WebID"
+        onChange={handleChange}
+        value={agentId}
+        required
```

Comment:
This makes the field invalid from the getgo - Although not optimal, I think we should not use this and rather have a test in the submit function that checks if there's a value before submitting (i.e. [what I did in this PR](https://github.com/inrupt/pod-browser/pull/100/commits/d0aa1b6135fd3c9ce21bc1bd48e79e01329d5f89#diff-1f62fb90f81f3d9cdb374658a11dc25e234a621460729589a97797d315f1fcccR51-R53)).